frontend/fingertips-frontend/lib/search/searchResultData.ts

```diff
@@ -0,0 +1,56 @@
+import { SearchService } from './searchService';
+
+export interface IndicatorSearchResult {
+  id: string;
+  indicatorName: string;
+  latestDataPeriod?: string;
+  dataSource?: string;
+  lastUpdated?: string;
+}
+
+export interface IndicatorSearch {
+  searchByIndicator(indicator: string): Promise<IndicatorSearchResult[]>;
+}
+
+export const MOCK_DATA: IndicatorSearchResult[] = [
+  {
+    id: '1',
+    indicatorName: 'NHS',
+    latestDataPeriod: '2023',
+    dataSource: 'NHS website',
+    lastUpdated: formatDate(new Date('December 6, 2024')),
+  },
+  {
+    id: '2',
+    indicatorName: 'DHSC',
+    latestDataPeriod: '2022',
+    dataSource: 'Student article',
+    lastUpdated: formatDate(new Date('November 5, 2023')),
+  },
+];
+
+function formatDate(date: Date): string {
+  const day = String(date.getDate()).padStart(2, '0');
+  const month = date.toLocaleString('en-GB', { month: 'long' });
+  const year = date.getFullYear();
+
+  return `${day} ${month} ${year}`;
+}
+
+let searchService: SearchService;
+try {
+  searchService = new SearchService();
```

Comment:
If I'm understanding this correctly. This will be a singleton of the SearchService instance on the NextJS server to handle all requests. Rather than create a new instance per request. Is this the intended idea here and does this have any draw backs? Just thinking performance, load or concurrency.

Reply:
Not sure, although I did think that performing the initial set-up of the search client on each call would be a repetition of effort. My thinking was if Node is single-threaded (think it might be) then creating a new instance each time doesn't help as it's just repeated work, & if it's multi-threaded then there's nothing to stop different threads running the same function.